lib/internal/crypto/cipher.js

```diff
@@ -86,10 +86,10 @@ function createCipherBase(cipher, credential, options, decipher, iv) {
   LazyTransform.call(this, options);
 }
 
-function throwInvalidArgType(name, value) {
+function throwInvalidArrayBufferView(name, value) {
   throw new ERR_INVALID_ARG_TYPE(
     name,
-    ['string', 'Buffer', 'TypedArray', 'DataView'],
```

Comment:
This change seems like a mistake. The `toBuf()` function accepts a string and transform that into a buffer.

Reply:
The motivation for this changes was that `toBuf()` would convert any string passed in to a `Buffer`, and therefore `isArrayBufferView` would not be passed a string for the `password`, `key`, and `iv` cases.

Note that my first commit also updated the `update` function which does not call `toBuf()` which was a mistake and this as fixed in the latest commit, so [update](https://github.com/nodejs/node/pull/22947/commits/bd3124678fc90de0f620b5ab382f1023425cedf6#diff-596ce0a0e2ea51999d5f982aaf07ff54R151-155) will also include the `string` in its throws cause.

Does that still sound incorrect?

Reply:
The error message corresponds to the possible values for the user, not to possible transformations we do internally before throwing the error. So `string` should still be valid in all of these cases.

Reply:
Sorry about the late reply. I see your point now, I'll revert my latest change. Thanks!